src/blockchain/log-processors/market-created.ts

```diff
@@ -1,18 +1,35 @@
-import { Database } from "sqlite3";
+import * as Knex from "knex";
 import { FormattedLog, ErrorCallback } from "../../types";
 
-export function processMarketCreatedLog(db: Database, log: FormattedLog, callback: ErrorCallback): void {
-  const dataToInsert: (string|number)[] = [
-    log.market, log.address, log.marketType, log.numOutcomes, log.minPrice, log.maxPrice, log.marketCreator, log.creationTime, log.blockNumber, log.creationFee, log.marketCreatorFeeRate, log.topic, log.tag1, log.tag2, log.reportingWindow, log.endTime, log.shortDescription, log.designatedReporter, log.resolutionSource
-  ];
-  db.run(`INSERT INTO markets
-    (market_id, universe, market_type, num_outcomes, min_price, max_price, market_creator, creation_time, creation_block_number, creation_fee, market_creator_fee_rate, topic, tag1, tag2, reporting_window, end_time, short_description, designated_reporter, resolution_source)
-    VALUES (${dataToInsert.map(() => '?').join(',')})`, dataToInsert, (err?: Error|null): void => {
+export function processMarketCreatedLog(db: Knex, log: FormattedLog, callback: ErrorCallback): void {
+  const dataToInsert: {} = {
+    contract_address:        log.market,
+    universe:                log.address,
+    market_type:             log.marketType,
+    num_outcomes:            log.numOutcomes,
+    min_price:               log.minPrice,
+    max_price:               log.maxPrice,
+    market_creator:          log.marketCreator,
+    creation_time:           log.creationTime,
+    creation_block_number:   log.blockNumber,
+    creation_fee:            log.creationFee,
+    market_creator_fee_rate: log.marketCreatorFeeRate,
+    topic:                   log.topic,
+    tag1:                    log.tag1,
+    tag2:                    log.tag2,
+    reporting_window:        log.reportingWindow,
+    end_time:                log.endTime,
+    short_description:       log.shortDescription,
+    designated_reporter:     log.designatedReporter,
+    resolution_source:       log.resolutionSource
+  };
+  db.insert(dataToInsert).into("markets").asCallback((err?: Error|null): void => {
       if (err) return callback(err);
-      db.get(`SELECT popularity FROM topics WHERE topic = ?`, [log.topic], (err?: Error|null, row?: {popularity: number}): void => {
+      db.raw(`SELECT popularity FROM topics WHERE topic = ?`, [log.topic]).asCallback((err?: Error|null, row?: {popularity: number}): void => {
```

Comment:
`asCallback` and `then` are inconsistently used throughout the code; I recommend picking one and sticking with it.  (If promise syntax is natural for knex I have no objection to its use.)

Reply:
Generally I am using asCallback except for all of the code which is interacting the already existing callback style APIs.

I use the promise interface (which indeed is natural in Knex) in the seeds and migrations.

I actually believe this is a natural way to approach combining the two interfaces, but I can move to the Promise-only if you prefer.

Reply:
Why not async/await?